app/modules/ads/slots.js

```diff
@@ -462,6 +464,43 @@ export const slots = {
       context.set('slots.incontent_boxad_1.targeting.xna', '0');
     }
   },
+
+  handleTopLeaderboardWrapper() {
+    const {
+      AdSlot,
+      context,
+      events,
+      eventService,
+      scrollListener,
+      slotService,
+    } = window.Wikia.adEngine;
+    const { universalAdPackage } = window.Wikia.adProducts;
+    const disableOnScroll = context.get('options.disableTopLeaderboardGapOnScroll');
+    const registerGapHandler = () => {
+      slotService.on('top_leaderboard', AdSlot.SLOT_RENDERED_EVENT, () => {
+        const adSlot = slotService.get('top_leaderboard');
+        adSlot.removeClass('is-loading');
+
+        if (universalAdPackage.isFanTakeoverLoaded()) {
+          adSlot.removeClass('wrapper-gap'); // shrink without animation
+        } else if (!disableOnScroll) {
+          adSlot.addClass('wrapper-gap-disabled'); // shrink with animation
+        } else {
+          const id = scrollListener.addCallback(() => {
+            if (!universalAdPackage.isFanTakeoverLoaded()) {
+              adSlot.addClass('wrapper-gap-disabled'); // shrink with animation
+
+              scrollListener.removeCallback(id);
+            }
+          });
+        }
```

Comment:
This part is completely unreadable for me. 🙈 
What do you think about creating separate methods like `shrinkWithouAnimation` and `shrinkWithoutAnimation` so there is no need to leave comments in code?
Also maybe we could move if statement from line 490 and make else if statement in 488?